### DISCUSSION TITLE
SeveritySource and LastModifiedDate

### DISCUSSION THREAD

**Discussion by sadunhaq123** [aquasecurity/trivy#4025] [Q&A] [answered]
Hello,

I would like to know where the SeveritySource and LastModifiedDate are pulled from. As for a vulnerability, if the source is 'redhat', do you pull the data for that vulnerability from Red Hat Security Advisory along with the 'Severity' and 'LastModifiedDate'? Or do you guys implement your own algorithms regarding what the 'Severity' should actually look like. Also the data for 'LastModifiedDate', is that also pulled from the source (here Red Hat Security Advisory), or is it the time you guys update the trivy database, or something else?

I am doing research on container security scanning and Trivy is one of the tools I am using. So any sort of reply will be highly appreciated.

Thank You.

**Comment by knqyf263** [comment, marked as answer]
It is listed here.
https://aquasecurity.github.io/trivy/v0.41/docs/scanner/vulnerability/os/#data-sources
